### PR TITLE
Rework UI search pages

### PR DESCRIPTION
**Pull request by joonatank**
## 🛠️ Changelog
[//]: # "Describe the changes in this pull request here."

- Combine code for both ui searches.
- Fix: type issues (remove most record types).
- Fix: use SSR to load the search page so it's rendered without a loading spinner.
- Fix: don't allow removing application rounds from the seasonal search.

## 🧪 Test plan
[//]: # "Help your fellow reviewer and write a short description of what's the fastest way to test your changes."

- Manual tests on both search pages (client application).

## 🎫 Tickets
[//]: # "This pull request resolves all or part of the following ticket(s)"

- [TILA-3241](https://helsinkisolutionoffice.atlassian.net/browse/TILA-3241)

[TILA-3241]: https://helsinkisolutionoffice.atlassian.net/browse/TILA-3241?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apps/ui/hooks/useSearchValues.tsx [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+        };
+      }
+    } else if (key) {
+      const { [`${key}`]: _, ...rest } = searchValues;
+      newValues = rest;
+    }
+
+    const { sort, order } = router.query;
+    const newSort = sort != null && !Array.isArray(sort) ? sort : null;
+    const newOrder = order != null && !Array.isArray(order) ? order : null;
```
Comment (joonatank) [inline apps/ui/hooks/useSearchValues.tsx:83]: IS this necessary?

File: apps/ui/modules/search.ts [hunk 2]
```diff
@@ -1,7 +1,20 @@
...
+  type QueryReservationUnitsArgs,
+  ReservationKind,
+  ReservationUnitOrderingChoices,
+} from "common/types/gql-types";
+import { ParsedUrlQuery } from "node:querystring";
+import { fromUIDate } from "./util";
+import { startOfDay } from "date-fns";
+import { toApiDate } from "common/src/common/util";
+
+const pagingLimit = 36;
```
Comment (joonatank) [inline apps/ui/modules/search.ts:17]: Move to const.

File: apps/ui/modules/search.ts [hunk 3]
```diff
@@ -78,3 +91,156 @@ export function transformSortString(
...
+): boolean | null {
+  if (param == null) return null;
+  if (param === "") return null;
+  if (Array.isArray(param)) {
+    return param.find((p) => p === "true") != null ? true : null;
+  }
+  return param === "true" ? true : null;
+}
+
+export function mapQueryParamToNumber(
```
Comment (joonatank) [inline apps/ui/modules/search.ts:234]: Naming??

**Review by vincit-matu — APPROVED** [review]
👍 👍

**Pull request merged**